geopy/geocoders/photon.py

```diff
@@ -116,16 +122,18 @@ def geocode(
                                   " coordinate pair or Point"))
         if osm_tag:
             if isinstance(osm_tag, string_compare):
-                params['osm_tag'] = osm_tag
+                encoded_osm_tag = urlencode(('osm_tag', osm_tag))
             else:
                 try:
-                    params['osm_tag'] = '&osm_tag='.join(osm_tag)
+                    encoded_osm_tag = urlencode(
+                        [('osm_tag', tag) for tag in osm_tag])
                 except ValueError:
                     raise ValueError(
                         "osm_tag must be a string expression or "
                         "a set/list of string expressions"
                     )
-        url = "?".join((self.api, urlencode(params)))
+        url = "?".join(
+            (self.api, "&".join((urlencode(params), encoded_osm_tag))))
```

Comment:
if `osm_tag` is a list you can get the same behaviour withour the encodes_osm_tag variable by just setting `doseq` to true in the call to urlenocde:

```python
params['osm_tag'] = osm_tag
...
url = "?".join(self.api, urlencode(params, doseq=True))
```